git-keeper.py

```diff
@@ -24,63 +24,82 @@
 tar = sh.tar.bake('-cf')
 workdir = 'workdir'
 
+
 def upload2s3(s3_client, repo_tar, git_keeper_bucket, date, object_name):
     try:
-        response = s3_client.upload_file(repo_tar, git_keeper_bucket, date + '/' + object_name)
+        s3_client.upload_file(
+            repo_tar, git_keeper_bucket, date + '/' + object_name)
     except ClientError as e:
         logging.error(e)
         sys.exit(1)
 
+
 def cleanwrkdir(workdir):
     shutil.rmtree(workdir, ignore_errors=True)
     os.makedirs(workdir, exist_ok=True)
 
+
 def clone_repo(repo_url, repo_dir):
     mirror(repo_url, repo_dir)
 
+
+def check_aws_creds(aws_access_key_id, aws_secret_access_key):
+    try:
+        boto3.client('s3',
+                     aws_access_key_id=aws_access_key_id,
+                     aws_secret_access_key=aws_secret_access_key)
+    except botocore.exceptions.NoCredentialsError:
+        raise Exception('No AWS credentials found.')
+    except botocore.exceptions.ClientError:
+        raise Exception('Invalid AWS credentials.')
+
+
+def git_clone_upload(s3_client, gpg, recipients, repo, s3_bucket, date):
+    repo_url = repo + '.git'
```

Comment:
I think this should be:

```
if not repo.endswith('.git'):
  repo += '.git'
```

Reply:
Sure, it's better. I just want in this PR deal with formatting and maybe moving to functions